### PR TITLE
test: add fuzzing case for the route with vars

### PR DESCRIPTION
**Pull request by idbeta**
### What this PR does / why we need it:
relate #3340
fuzzing for the route with vars

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by idbeta** [comment]
ping @spacewander can you check the CI fail? It seems that the case is unstable.

**Comment by spacewander** [comment]
It is the solved problem I said at the noon.

File: t/fuzzing/vars_route_test.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    session.fuzz()
+
+if __name__ == "__main__":
+    # before test
+    create_route()
+    r1 = check_process()
+    main()
+    # after test
+    boofuzz_log = cur_dir() + "/test.log"
+    apisix_errorlog = "~/work/apisix/apisix/logs/error.log"
```
Comment (spacewander) [inline t/fuzzing/vars_route_test.py:89]: All the fuzzy scripts are similar. Can we share the common part of them?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: t/fuzzing/vars_route_test.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import subprocess
+from public import *
+from boofuzz import *
```
Comment (moonming) [inline t/fuzzing/vars_route_test.py:22]: “import * ” is not good style in python

**Review by moonming — CHANGES_REQUESTED** [review]

File: t/fuzzing/vars_route_test.py (hunk 1 above)
Comment (idbeta) [reply t/fuzzing/vars_route_test.py:89]: yes, let me fixed it.

**Comment by idbeta** [comment]
> It is the solved problem I said at the noon.
> Please merge master to make CI pass.

Thanks for remind.

File: t/fuzzing/serverless_route_test.py [hunk 3]
```diff
@@ -100,12 +92,10 @@ def main():
     # before test
```
Comment (spacewander) [inline t/fuzzing/serverless_route_test.py:92]: Can we refactor this part as function and use `create_route` & `main` as arguments?

File: t/fuzzing/vars_route_test.py (hunk 2 above)
Comment (idbeta) [reply t/fuzzing/vars_route_test.py]: `from public import *` —— I have changed.
`from boofuzz import *` ——This is the official [usage](https://github.com/jtpereyda/boofuzz/blob/master/examples/http_simple.py),  I think it's ok.

File: t/fuzzing/serverless_route_test.py (hunk 3 above)
Comment (idbeta) [reply t/fuzzing/serverless_route_test.py]: I got it.

File: t/fuzzing/public.py [hunk 4]
```diff
@@ -43,3 +44,27 @@ def check_process():
...
+    session = Session(
+        target=Target(
+            connection=TCPSocketConnection("127.0.0.1", 9080, send_timeout=5.0, recv_timeout=5.0, server=False)
+        ),
+        fuzz_loggers=fuzz_loggers,
+        keep_web_open=False,
+    )
+    return session
+
+def runtest(create_route, run):
```
Comment (spacewander) [inline t/fuzzing/public.py:60]: Should be `run_test`?

**Review by spacewander — APPROVED** [review]

File: t/fuzzing/public.py (hunk 4 above)
Comment (idbeta) [reply t/fuzzing/public.py:60]: sure

**Comment by idbeta** [comment]
ping @moonming

File: t/fuzzing/public.py [hunk 5]
```diff
@@ -20,14 +20,15 @@
 import subprocess
 import os
 import psutil
+from boofuzz import *
```
Comment (moonming) [inline t/fuzzing/public.py:23]: still not fixed

Comment (idbeta) [reply t/fuzzing/public.py:23]: This is the official usage. If it is changed to `import boofuzz`, the code will be very complicated. @moonming

**Review by moonming — APPROVED** [review]

**Pull request merged**